sympy/solvers/ode/riccati.py

```diff
@@ -0,0 +1,506 @@
+    return c
+
+
+def construct_d(num, den, x, val_inf, mul):
+    N = -val_inf//2
+    ser = rational_laurent_series(num, den, x, oo, mul, 1)
+
+    # Case 4
+    if val_inf < 0:
+        temp = [0 for i in range(N+2)]
+        temp[N] = sqrt(ser[-mul - 1 + 2*N])
+        for s in range(N-1, -2, -1):
+            sm = 0
+            for j in range(s+1, N):
+                sm += temp[j]*temp[N+s-j]
+            if s != -1:
+                temp[s] = (ser[-mul - 1 + N+s] - sm)/(2*temp[N])
+        temp1 = [-x for x in temp]
+        # The third equation in Eq 5.15 of the thesis is WRONG!
+        # d_N must be replaced with N*d_N in that equation.
+        temp[-1] = (ser[-mul - 1 + N+s] - N*temp[N] - sm)/(2*temp[N])
+        temp1[-1] = (ser[-mul - 1 + N+s] - N*temp1[N] - sm)/(2*temp1[N])
+        d = [temp, temp1]
+
+    # Case 5
+    elif val_inf == 0:
+        # List to store coefficients for plus case
+        temp = [0, 0]
+
+        # d_0  = sqrt(a_0)
+        temp[0] = sqrt(ser[-mul - 1])
+
+        # d_(-1) = a_(-1)/(2*d_0)
+        temp[-1] = ser[-mul - 2]/(2*temp[0])
+
+        # Coefficients for the minus case are just the negative
+        # of the coefficients for the positive case. Append both.
+        d = [temp, [-x for x in temp]]
+
+    # Case 6
+    else:
+        # s_oo = lim x->0 1/x**2 * a(1/x) which is equivalent to
+        # s_oo = lim x->oo x**2 * a(x)
+        s_inf = limit_at_inf(Poly(x**2, x)*num, den, x)
+
+        # d_(-1) = (1 +- sqrt(1 + 4*s_oo))/2
+        d = [[(1 + sqrt(1 + 4*s_inf))/2], [(1 - sqrt(1 + 4*s_inf))/2]]
+    return d
+
+
+def rational_laurent_series(num, den, x, x0, mul, n):
+    one = Poly(1, x)
+    reverse = False
+    if x0 == oo:
+        num, den = inverse_transform_poly(num, den, x)
+        reverse = True
+        x0 = 0
+    if x0 != 0:
+        num = num.transform(Poly(x + x0, x), one)
+        den = den.transform(Poly(x + x0, x), one)
+        num, den = num.cancel(den, include=True)
+    num, den = (num*x**mul).cancel(den, include=True)
+
+    coeff_max = max(den.degree(x), num.degree(x) + 1)
+    c = list(symbols(f'c:{coeff_max}', cls=Dummy))
+    out = Poly(0, x, domain=ZZ[c])
+    rcoeffs = []
+
+    for pw, cf in den.as_dict().items():
+        pw = pw[0]
+        for i in range(pw, coeff_max):
+            out += Poly(cf*c[i - pw]*x**i, x, domain=ZZ[c])
+        rcoeffs.append((cf, -pw))
+
+    sols = list(linsolve((num - out).all_coeffs(), c, x))
+    if len(sols):
+        c[:len(sols)] = sols[0]
+        for i in range(len(sols), n + mul):
+            sums = sum([cf*c[i - pw] for cf, pw in rcoeffs])
+            sol = list(linsolve([sums], [c[i]], x))
+            if len(sol):
+                c[i] = sol[0][0]
+    for i in range(len(c)):
+        if len(S(c[i]).atoms(Dummy)):
+            c = c[:i]
+            break
+    return c[::-1] if reverse else c
+
+
+def compute_degree(x, poles, choice, c, d, N):
+    ybar = 0
+    m = Poly(d[choice[0]][-1], x, extension=True)
```

Comment:
If this calculates `m` and `ybar` then I think that the name should be something like `compute_m_ybar`.